File: sunnad-ios/sunnad-ios/Domain/Repositories/GroupsRepository.swift
import Foundation

protocol GroupsRepository: Sendable {
    func fetchGroups() async throws -> [Group]
    func createGroup(name: String) async throws -> Group
    func joinGroup(code: String) async throws -> Group
    func renameGroup(groupID: UUID, name: String) async throws
    func setJoinLock(groupID: UUID, locked: Bool) async throws
    func rotateInviteCode(groupID: UUID) async throws -> String
    func updateSharing(groupID: UUID, habitIDs: Set<UUID>) async throws
    func leaveGroup(groupID: UUID) async throws
    func deleteGroup(groupID: UUID) async throws
    func kickMember(groupID: UUID, memberUserID: UUID) async throws
    func setProgressDisplayMode(groupID: UUID, mode: GroupProgressDisplayMode) async throws
    func refreshGroup(groupID: UUID) async throws -> Group?
    func sendNudge(groupID: UUID, toUserID: UUID, habitID: UUID) async throws -> GroupNudgeStatus
}

File: sunnad-android/app/src/main/java/com/arystan/almasuly/sunnadandroid/domain/repository/GroupsRepository.kt
package com.arystan.almasuly.sunnadandroid.domain.repository

import com.arystan.almasuly.sunnadandroid.core.model.Group
import com.arystan.almasuly.sunnadandroid.core.model.GroupNudgeStatus
import com.arystan.almasuly.sunnadandroid.core.model.GroupProgressDisplayMode
import java.util.UUID

interface GroupsRepository {
    suspend fun fetchGroups(): List<Group>
    suspend fun createGroup(name: String): Group
    suspend fun joinGroup(code: String): Group
    suspend fun renameGroup(groupId: UUID, name: String)
    suspend fun setJoinLock(groupId: UUID, locked: Boolean)
    suspend fun rotateInviteCode(groupId: UUID): String
    suspend fun updateSharing(groupId: UUID, habitIds: Set<UUID>)
    suspend fun leaveGroup(groupId: UUID)
    suspend fun deleteGroup(groupId: UUID)
    suspend fun kickMember(groupId: UUID, memberUserId: UUID)
    suspend fun setProgressDisplayMode(groupId: UUID, mode: GroupProgressDisplayMode)
    suspend fun refreshGroup(groupId: UUID): Group?
    suspend fun sendNudge(groupId: UUID, toUserId: UUID, habitId: UUID): GroupNudgeStatus
}
